src/Security/Items.cs

```diff
@@ -1733,4 +1733,65 @@ public SecurityException (SecStatusCode code) : base (ToMessage (code))
 		{
 		}
 	}
+
+	public partial class SecKeyParameters : DictionaryContainer {
+		// For caching, as we can't reverse it easily.
+		SecAccessControl _secAccessControl;
+
+		public SecAccessControl AccessControl {
+			get {
+				return _secAccessControl;
+			}
+			set {
+				if (value == null)
+					throw new ArgumentNullException (nameof (value));
+				_secAccessControl = value;
+				SetNativeValue (SecAttributeKeys.AccessControlKey, value);
+			}
+		}
+	}
+
+	public partial class SecKeyGenerationParameters : DictionaryContainer {
+		public SecKeyType KeyType {
+			get {
+				var type = GetNSStringValue (SecKeyGenerationAttributeKeys.KeyTypeKey);
+				if (type == null)
+					return SecKeyType.Invalid;
+				return SecKeyTypeExtensions.GetValue (type);
+			}
+
+			set {
+				var k = value.GetConstant ();
+				if (k == null)
+					throw new ArgumentException ("Unknown value for KeyType.");
+				SetStringValue (SecKeyGenerationAttributeKeys.KeyTypeKey, k);
+			}
+		}
+
+		// For caching, as we can't reverse it easily.
+		SecAccessControl _secAccessControl;
+
+		public SecAccessControl AccessControl {
+			get {
+				return _secAccessControl;
+			}
+
+			set {
+				if (value == null)
+					throw new ArgumentNullException (nameof (value));
+				_secAccessControl = value;
+				SetNativeValue (SecAttributeKeys.AccessControlKey, value);
+			}
+		}
+
+		public SecTokenID TokenID {
```

Comment:
I think this should be `TokenId` instead

Reply:
yes, but let's be consistent with existing usages